### ISSUE TITLE
Token authentication using bearer scheme

### ISSUE THREAD

**Issue by DirkMahler** [bsorrentino/maven-confluence-plugin#280]
I'm in a corporate environment where I need to publish documents to a confluence instance using an authentication token. The documentation of the Maven plugin states that this is possible specifying user credentials in the server settings using the token as password. Sadly the server does not accept the basic auth scheme for tokens, it requires the Bearer scheme instead. 

It would be good if the used authentication scheme could either be configured explicitly (if "bearer" then just use the configured password) or headers could be specified for confluence instances in the server section of settings.xml (similar to https://maven.apache.org/guides/mini/guide-http-settings.html and https://github.com/maven-download-plugin/maven-download-plugin/issues/163).

If the solution direction is clear then I could spend some time to prepare a PR.

**Comment by bsorrentino** [comment]
Hi @DirkMahler 

Thanks for feedback. Until now I never deal with this issue. In such case a PR is absolutely welcome

**Comment by DirkMahler** [comment]
I've created a PR (https://github.com/bsorrentino/maven-confluence-plugin/pull/281) containing changes regarding the REST API. Authentication is now handled using an interceptor of OkHttp, credentials and/or HTTP headers are taken from the settings.xml (following the approach described by Maven):

```
    <server>
      <id>confluence</id>
      <configuration>
        <httpHeaders>
          <property>
            <name>Authentication</name>
            <value>Bearer xxxxxxx</value>
          </property>
        </httpHeaders>
      </configuration>
    </server>       
```

Can you have a look if the changes make sense to you? Still open are documentation updates and tests.

**Comment by bsorrentino** [comment]
Hi @dirkmahler

I am amazed by your effectiveness, I will review as soon as possible ... but I guess I will be slow considering your speed

**Comment by bsorrentino** [comment]
fix on release [7.8](https://github.com/bsorrentino/maven-confluence-plugin/releases/tag/v7.8)
